jre_emul/Classes/dalvik/system/VMStack.java

```diff
@@ -14,11 +16,101 @@
 
 package dalvik.system;
 
+import android.compat.annotation.UnsupportedAppUsage;
+
+import dalvik.annotation.optimization.FastNative;
+
 /**
- * Stubbed version of Android's VMStack.
+ * Provides a limited interface to the Dalvik VM stack. This class is mostly
+ * used for implementing security checks.
+ *
+ * @hide
  */
+@libcore.api.CorePlatformApi
 public final class VMStack {
-  public static ClassLoader getCallingClassLoader() {
-    return null;
+
+  private VMStack() {
   }
+
+  /**
+   * Returns the defining class loader of the caller's caller.
+   *
+   * @return the requested class loader, or {@code null} if this is the
+   *         bootstrap class loader.
+   * @deprecated Use {@code ClassLoader.getClassLoader(sun.reflect.Reflection.getCallerClass())}.
+   *         Note that that can return {@link BootClassLoader} on Android where the RI
+   *         would have returned null.
+   */
+  @UnsupportedAppUsage
+  @FastNative
+  @Deprecated
+  native public static ClassLoader getCallingClassLoader();
```

Comment:
There needs to be a matching native code implementation for this to have a static modifier. However, since all classes are "boot loaded" on j2objc (since we cannot support loading external code), let's use the old non-native implementation from line 22 (just return null), moved here for the doc-comment and annotations.